src/main/java/ro/msg/learning/shop/ShopApplication.java

```diff
@@ -2,12 +2,35 @@
 
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.context.annotation.Bean;
+import org.springframework.transaction.annotation.EnableTransactionManagement;
+import ro.msg.learning.shop.model.Customer;
+import ro.msg.learning.shop.repository.CustomerRepository;
 
 @SpringBootApplication
+@EnableTransactionManagement
 public class ShopApplication {
 
 	public static void main(String[] args) {
 		SpringApplication.run(ShopApplication.class, args);
 	}
 
+	/**
+	 * Returns a customer from the database, or a dummy customer if there is none
+	 *
+	 * @param customerRepository
+	 * @return
+	 */
+	@Bean
+	Customer customer(CustomerRepository customerRepository) {
```

Comment:
I understand what you want to do with this bean, and it's a good idea. But let's move it elsewhere, in a configuration file, just to be separated from the Application. You have more details here: https://docs.spring.io/spring-javaconfig/docs/1.0.0.m3/reference/html/creating-bean-definitions.html.

Reply:
I moved it to a separate configuration file :)